presentation/src/main/java/com/wednesday/template/presentation/weather/search/SearchViewModel.kt

```diff
@@ -41,8 +43,24 @@ class SearchViewModel(
     override fun onCreate(fromRecreate: Boolean) {
 
         searchCityInteractor.searchResultsFlow.onEach {
-            setState {
-                copy(showLoading = false, searchList = it)
+            when (it) {
+                is UIResult.Success -> {
+                    setState {
+                        copy(showLoading = false, searchList = it.data)
+                    }
+                }
+                is UIResult.Error -> {
+                    setState {
+                        copy(showLoading = false)
+                    }
+                    setEffect(
+                        ShowSnackbarEffect(
+                            message = UIText {
+                                block(it.exception.message ?: "Something went wrong, Please try again")
```

Comment:
Let put the string in strings.xml

Reply:
The problem is we can't have context in the viewmodel and then no access to `context.getString(R.string.xyz`

Reply:
> The problem is we can't have context in the viewmodel and then no access to `context.getString(R.string.xyz`

@shounak-mulay

Reply:
@gtxtreme Let's use the resource id here as well.